src/main/java/net/glowstone/block/PistonMoveBehavior.java

```diff
@@ -0,0 +1,12 @@
+package net.glowstone.block;
+
+/**
+ * Behavior of a block when moved (pushed/pulled) py a piston.
+ *
+ * <p>MOVE - move as usual.</p>
+ * <p>BREAK - break block and drop necessary items (e.g. flowers).</p>
+ * <p>DONT_MOVE - don't move the block (e.g. bedrock).</p>
+ */
```

Comment:
Doesn't the Slime Block need a fourth mode?

Reply:
What mode are you talking about? Special mode to determine if the block is slime? I don't know then, there is only one block with such properties for now.

And the Wiki says this:

> The types of blocks that can be moved are the same as those that can be pulled by a sticky piston. Blocks that cannot be pulled by a sticky piston (i.e. all the blocks listed on the table on the pistons page) stay in place.' 

So, I think that pistonPullBehavior property is enough for that.

Reply:
I mean to implement the behavior where a slime block moves other blocks when moved by a piston. 
https://gamepedia.cursecdn.com/minecraft_gamepedia/9/94/SlimeBlockFlyingMachineAnimation300lrgr.gif?version=b8b2274f33e41ad881e1878965f314de

Reply:
Isn't it enough to just check if the material is slime block? So another variant in the enum is not needed.

I understand about the slime logic itself and I'm gonna work on it.

Reply:
I would encourage you to rethink hard coding the slime block's "sticky" behavior. Even though slime blocks are the only instance of this behavior today doesn't mean it'll always be that way. Shouldn't be too much more work to abstract this behavior into some form of flag.

Reply:
Isn't there glazed terracotta had a special case that it could only be pushed but not pulled?

Reply:
Isn't that covered by separate pull and push behavior?

Reply:
@garantiertnicht I've got separate move behaviors for pushing and pulling so it's not a problem. We can set Terracota's pushing behavior to MOVE and pulling behavior to DONT_MOVE.